src/pages/VenuePage/VenuePage.tsx

```diff
@@ -204,12 +318,25 @@ export const VenuePage: React.FC = () => {
     venue.hasPaidEvents &&
     !isUserVenueOwner
   ) {
-    if (eventRequestStatus && !event) {
-      return <>This event does not exist</>;
+    if (isEventLoaded && !event) {
+      return (
+        <>
+          <Preload venue={venue} />
+          This event does not exist
+        </>
+      );
     }
 
     if (!event || !venue || !userPurchaseHistoryRequestStatus) {
-      return <LoadingPage />;
+      // considering there is prior !venue check, venue should exist at this point
```

Comment:
Can probably be removed then?

Reply:
done 6e1ca9a6086f3a008b5e180b5c0f676ffc041046